components/org.wso2.carbon.identity.api.idle.account.identification/org.wso2.carbon.identity.api.idle.account.identification.v1/src/main/java/org/wso2/carbon/identity/api/idle/account/identification/v1/core/InactiveUsersManagementApiService.java

```diff
@@ -0,0 +1,243 @@
+        ErrorMessage error = ErrorMessage.ERROR_DATE_REGEX_MISMATCH;
+        throw new IdleAccountIdentificationClientException(error.getCode(), error.getMessage(),
+                String.format(error.getDescription(), dateType));
+    }
+
+    /**
+     * Convert date string into LocalDateTime object.
+     *
+     * @param dateString Date as a string.
+     * @param dateType   Date type.
+     * @throws IdleAccountIdentificationClientException IdleAccIdentificationClientException.
+     * @return List of inactive users.
+     */
+    private LocalDateTime convertToDateObject(String dateString, String dateType)
+            throws IdleAccountIdentificationClientException {
+
+        try {
+            if (StringUtils.isEmpty(dateString)) {
+                return null;
+            }
+            return LocalDate.parse(dateString).atStartOfDay();
+        } catch (DateTimeParseException e) {
+            ErrorMessage error = ErrorMessage.ERROR_INVALID_DATE;
+            throw new IdleAccountIdentificationClientException(error.getCode(), error.getMessage(),
+                    String.format(error.getDescription(), dateType));
+        }
+    }
+
+    /**
+     * Build the InactiveUser list.
+     *
+     * @param inactiveUserModels List of inactive users.
+     * @return List of inactive users.
+     */
+    private List<InactiveUser> buildResponse(List<InactiveUserModel> inactiveUserModels) {
+
+        List<InactiveUser> inactiveUserList = new ArrayList<>();
+        for (InactiveUserModel inactiveUserModel : inactiveUserModels) {
+            InactiveUser inactiveUser = new InactiveUser();
+            inactiveUser.setUsername(inactiveUserModel.getUsername());
+            inactiveUser.setUserStoreDomain(inactiveUserModel.getUserStoreDomain());
+            inactiveUser.setUserId(inactiveUserModel.getUserId());
+            inactiveUserList.add(inactiveUser);
+        }
+        return inactiveUserList;
+    }
+
+    /**
+     * Handle IdleAccIdentificationException.
+     *
+     * @param exception IdleAccIdentificationException.
+     * @param errorEnum Error message.
+     * @param data      Context data.
+     * @return APIError.
+     */
+    private APIError handleIdleAccIdentificationException(IdleAccountIdentificationException exception,
+                                                          IdleAccountIdentificationConstants.ErrorMessage errorEnum,
+                                                          String data) {
+
+        ErrorResponse errorResponse;
+        Response.Status status;
+        if (exception instanceof IdleAccountIdentificationClientException) {
+            errorResponse = getErrorBuilder(errorEnum, data).build(LOGGER, exception.getMessage());
+            if (exception.getErrorCode() != null) {
+                errorResponse.setCode(exception.getErrorCode());
+            }
+            errorResponse.setDescription(exception.getMessage());
+            errorResponse.setRef(getCorrelation());
```

Comment:
There are few common lines like this within all the conditions. Extract them to a common place.

Reply:
Addressed with commit e52d2135a2cab52434b3a4ae45415981e1ab9011